File: _practice/2181-Merge-Nodes-in-Between-Zeros/sam/in_place.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var mergeNodes = function (head) {
  head = head.next;
  let tail = haed;
  while (tail) {
    let cur = tail;
    while (cur.next.val !== 0) {
      tail.val += cur.next.val;
      cur = cur.next;
    }
    tail.next = cur.next.next;
    tail = tail.next;
  }

  return head;
};

File: _practice/2181-Merge-Nodes-in-Between-Zeros/sam/in_place.py
# Definition for singly-linked list.
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next
class Solution:
    def mergeNodes(self, head):
        head = head.next
        tail = head
        while tail:
            cur = tail
            while cur.next.val != 0:
                tail.val += cur.next.val
                cur = cur.next
            tail.next = cur.next.next
            tail = tail.next          

        return head